scripts/extraction/wikidata.py

```diff
@@ -0,0 +1,324 @@
+""" Functionalities for processing Wikidata dump.
+Modified from https://github.com/explosion/projects/blob/master/nel-wikipedia/wikidata_processor.py.
+"""
+
+import bz2
+import io
+import json
+import multiprocessing
+import sqlite3
+from pathlib import Path
+from typing import Union, Optional, Dict, Tuple, Any, List, Set
+
+import tqdm
+
+from .utils import get_logger
+from .namespaces import WD_META_ITEMS
+
+
+def _read_dump(
```

Comment:
Splitting extraction into one reading and one parsing process reduces the extraction time for Wikidata by around 40%.